### PR TITLE
21206 Far grab improvement to draw held entity toward yourself

### PR DESCRIPTION
**Pull request by volansystech**
**Objective:** 
- Improve speed & experience of far Grabbing. Also drag the object as close as possible to Avatar.

**What to test?**
- Do far grab from some long distance for all entities and check "Object travelling speed".

**Expected behavior:**
- When we draw object from far grab range, the speed at which object travels to the Avatar is improved.
- "Object travelling speed" will vary as per entity mass and distanceToObject.

- We tested on following objects in welcome-dev area and it is working as expected.
	1) EZ-Cube
	2) EZ-Blok
	3) EZ-Tube
	4) EZ-Arch
	5) EZ-Ramp
	6) Cylinder shape object of palace
	7) Heaviest Cube
	
@themelissabrown , Please ask QA To Test.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
An admin must verify this patch for the pull request build to fire. Type "ok to test" to begin testing this pull request or "add to whitelist" to test this and all future pull requests from this collaborator.

**Comment by themelissabrown** [comment]
ok to test

**Comment by hifi-gustavo** [comment]
Build console output for commit: e5c7587ad8b36f16f108098a5cadd2ae5eec04d4 available at: https://gist.github.com/c6aa1679a4d388a469d3e67223dea9b3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10235-e5c7587.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10235-e5c7587.dmg

**Comment by volansystech** [comment]
@themelissabrown ,Please mark this PR as **"QA Ready"** so someone can test it.

**Comment by claudioshg** [comment]
testing

File: scripts/system/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -2661,9 +2658,13 @@ function MyController(hand) {
                            this.grabbedThingID);
 
         var distanceToObject = Vec3.length(Vec3.subtract(MyAvatar.position, this.currentObjectPosition));
+        this.linearTimeScale = (this.linearTimeScale / 2);
+        if(this.linearTimeScale <= DISTANCE_HOLDING_ACTION_TIMEFRAME){
```
Comment (sethalves) [inline scripts/system/controllers/handControllerGrab.js]: space after if

**Comment by claudioshg** [comment]
### Desktop PC - Windows 10 Home - CPU 4.0GHz Intel i7 - GTX1070 - RAM 16GB - Tested with Oculus Rift

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10235-e5c7587.exe

PASS:
- Speed has increased for entities when traveling from a long distance compared to previous builds.
- All 7 EZ entities behaved as expected.
- Tested distance using ground markings. 
- At 80m all entities took 11 seconds to travel from point A to point B. 
- Testing in this way made me realize that all entities travel at the same speed. Even the Heavy Cube traveled at the same speed as a normal cube. Weight didn't seem to factor.
- Distance does affect entity travel speeds. The closer one is to an EZ-cube the faster it travels to you. The closer one is to an EZ-Heavy Cube, it's weight seems to affect its speed, therefore travels slower to you. However, at far distances, all EZ-entities traveled at the same speed.

File: scripts/system/controllers/handControllerGrab.js (hunk 1 above)
Comment (volansystech) [inline scripts/system/controllers/handControllerGrab.js]: Done. Thanks.

**Comment by hifi-gustavo** [comment]
Build console output for commit: b194ddeb66d4552270e3894710cfc9166fbae2a4 available at: https://gist.github.com/c6aa1679a4d388a469d3e67223dea9b3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10235-b194dde.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR10235-b194dde.dmg

**Pull request merged**